### PR TITLE
Feature/modeling entities

### PR DESCRIPTION
**Pull request by KostiukovAleksey**
basic entities added as well as controllers/services...

### CODE REVIEW

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/DTOs/AddressDTO.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.nokia.library.nokiainnovativeproject.DTOs;
+
+import lombok.*;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class AddressDTO {
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/DTOs/AddressDTO.java]: There are no constraints on the fields in any DTO. Now there is possibility to save nulls to database. 
Also in every Controller this constraints should be checked.

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.User;
+
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig  extends WebSecurityConfigurerAdapter {
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/config/SecurityConfig.java]: Why security conf is added in entities feature? 
I think that you do not need it here. This configuration can be done in separate feature.

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookDetailsController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		return new MessageInfo(true, bookDetailsService.createBookDetails(bookDetailsDTO), "bookDetails created successfully");
+	}
+
+	@PostMapping(Mappings.UPDATE)
+	public MessageInfo updateBook(@PathVariable Long id, @RequestBody BookDetailsDTO bookDetailsDTO) {
+		return new MessageInfo(true, bookDetailsService.updateBookDetails(id, bookDetailsDTO), "bookDetails updated successfully");
+	}
+
+	@DeleteMapping(Mappings.REMOVE)
+	public MessageInfo deleteBook(@PathVariable Long id) throws ResourceNotFoundException {
```
[the remark this answers is not in the document]
Comment (tubidubidam) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookDetailsController.java]: it's a big assumption...
but perhaps, yes :)

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/User.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import javax.persistence.*;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+
+@Entity
+@Table(name="\"User\"")
```
[the remark this answers is not in the document]
Comment (tubidubidam) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/User.java]: no, it's forbidden word, we had to set it manually :)
https://stackoverflow.com/questions/10891368/postgres-table-column-name-restrictions
and docs: https://www.postgresql.org/docs/current/sql-keywords-appendix.html

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Address.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.io.Serializable;
+
+@Entity
+@Getter
+@EqualsAndHashCode
+@NoArgsConstructor
```
[the remark this answers is not in the document]
Comment (tubidubidam) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Address.java]: that would be correct, but @Data annotation has @RequiredArgsConstrucot in it, so if we need 0 arguments constructor, we have to add it separately 
https://projectlombok.org/features/Data

File: backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/controllers/BookControllerTest.java [hunk 6]
```diff
@@ -24,87 +22,107 @@
...
+    }
+
+    @BeforeEach
+    public void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(controller).build();
+    }
+
+    @Test
+    public void getBooksTest() throws Exception {
+        List<Book> books = new ArrayList<>();
```
Comment (MateuszPe) [inline backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/controllers/BookControllerTest.java]: You can use `asList()` method :)

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/DTOs/ReviewDTO.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.Date;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class ReviewDTO {
+
+    private String comment;
+    private Date addDate;
```
[the remark this answers is not in the document]
Comment (MateuszPe) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/DTOs/ReviewDTO.java]: Please ignore the above comment. The comment should be attached to Review entity, not to DTO.

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookController.java [hunk 8]
```diff
@@ -1,44 +1,46 @@
...
-			throws ResourceNotFoundException {
+	@PostMapping(Mappings.UPDATE)
+	public MessageInfo updateBook(@PathVariable Long id, @RequestBody @Valid BookDTO bookDTO){
+		return new MessageInfo(true, bookService.updateBook(id, bookDTO), "Book updated successfully");
+	}
+
+	@DeleteMapping(Mappings.REMOVE)
+	public MessageInfo deleteBook(@PathVariable Long id) throws ResourceNotFoundException {
 		bookService.deleteBook(id);
+		return new MessageInfo(true, null, "Book with ID = " + id.toString() + " removed successfully");
```
[the remark this answers is not in the document]
Comment (tubidubidam) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookController.java]: we came up with the idea of creating 2 constructors, 1 for success situation and 2nd for any kind of failure, it's not like we dont like `@Builder`, but we don't want to overcomplicate things :)

msginf

[the remark this answers is not in the document]
Comment (MateuszPe) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/controllers/BookController.java]: You can also create static method for success: 
```
MessageInfo success(Object object, List<String> messages) {
   return new MessageInfo(true, object, messages);
}
And second one for failures. 
```

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Address.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Dasiuss) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Address.java]: but there are 0 required arguments in most of entity classes, so requiredArgsConstructor should generate 0arg constructor :P

**Pull request merged**